src/Controller/AdminController.php

```diff
@@ -2,472 +2,534 @@
     private $entityManager;
+    private $paginator;
+    private $translator;
 
     /**
      * AdminController constructor.
+     * @param \App\Repository\CategoryRepository $categoryRepository
+     * @param \App\Repository\ThemeCategoryRepository $themeCategoryRepository
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     * @param \Knp\Component\Pager\PaginatorInterface $paginator
+     * @param \Symfony\Contracts\Translation\TranslatorInterface $translator
      */
-    public function __construct(CategoryRepository $categoryRepository, ThemeCategoryRepository $themeCategoryRepository, EntityManagerInterface $entityManager)
-    {
+    public function __construct(
+        CategoryRepository $categoryRepository,
+        ThemeCategoryRepository $themeCategoryRepository,
+        EntityManagerInterface $entityManager,
+        PaginatorInterface $paginator,
+        TranslatorInterface $translator
+    ) {
         $this->categoryRepository = $categoryRepository;
         $this->themeCategoryRepository = $themeCategoryRepository;
         $this->entityManager = $entityManager;
+        $this->paginator = $paginator;
+        $this->translator = $translator;
     }
 
     /**
-     * Serves dashboard.
+     * Overrides.
      *
-     * @return mixed
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
      */
-    protected function dashboardAction()
+    public function editAction()
     {
-        $this->dispatch(EasyAdminEvents::PRE_LIST);
-
-        $query = trim($this->request->query->get('query'));
-
-        $paginator = null;
-
-        $maxNumberOfResults = 15;
-
-        if ('' === $query) {
-            $paginator = $this->findAll(
-                $this->entity['class'],
-                $this->request->query->get('page', 1),
-                $maxNumberOfResults,
-                'id',
-                'ASC',
-                $this->entity['list']['dql_filter']
-            );
-        } else {
-            $this->entity['search']['fields'] = ['name' => $this->entity['search']['fields']['name']];
-
-            $searchableFields = $this->entity['search']['fields'];
-
-            $paginator = $this->findBy(
-                $this->entity['class'],
-                $query,
-                $searchableFields,
-                $this->request->query->get('page', 1),
-                $maxNumberOfResults,
-                isset($this->entity['search']['sort']['field']) ? $this->entity['search']['sort']['field'] : $this->request->query->get(
-                    'sortField'
-                ),
-                'ASC',
-                $this->entity['search']['dql_filter']
-            );
+        $entityArray = $this->entity;
+        switch ($entityArray['class']) {
+            case Report::class:
+            case System::class:
+                $entity = $this->entityManager->getRepository($entityArray['class'])->find($_GET['id']);
```

Comment:
A `getEntity` function may be useful here. 
Apparently, EasyAdminController (also) has an EntityManager instance, cf. https://symfony.com/doc/master/bundles/EasyAdminBundle/book/complex-dynamic-backends.html#admincontroller-properties-and-methods.